core/dblookupext/historyRepository.go

```diff
@@ -50,6 +50,7 @@ type historyRepository struct {
 	// that could mistakenly override the "patch()" operations performed when consuming notarization notifications.
 	deduplicationCacheForInsertMiniblockMetadata storage.Cacher
 
+	// Question for review: this can be removed now, using the new approach (without the block tracker), right?
```

Comment:
The only situation when the notifications will be redundant will be when a node will do a rollback. In this situation the same final block/blocks will be notified again, after another new block will be constructed upon it. But this case should not be a regular pattern and could be considered as an exception. If this redundant notifications will not break something in the implementation logic and the below method was used just as an optimization, I would say that you can remove it now, considering the new approach.

Reply:
The implementation of `OnNotarizedBlocks()` is all right with being called twice with some redundant data in case of rollbacks.